src/ui/ResultList/ResultList.ts

```diff
@@ -320,27 +339,32 @@ export class ResultList extends Component {
     $$(this.root).on(ResultListEvents.changeLayout, (e: Event, args: IChangeLayoutEventArgs) => this.handleChangeLayout(args));
 
     if (this.options.enableInfiniteScroll) {
-      this.handlePageChanged();
-      this.bind.on(<HTMLElement>this.options.infiniteScrollContainer, 'scroll', (e: Event) => {
-        this.successiveScrollCount = 0;
-        this.handleScrollOfResultList();
-      });
+      this.addInfiniteScrollListeners();
+    } else {
+      this.addPagerListeners();
     }
-    this.bind.onQueryState(MODEL_EVENTS.CHANGE_ONE, QUERY_STATE_ATTRIBUTES.FIRST, () => this.handlePageChanged());
-
-    this.resultContainer = this.initResultContainer();
-    Assert.exists(this.options.resultsContainer);
+  }
 
-    this.initWaitAnimationContainer();
-    Assert.exists(this.options.waitAnimationContainer);
+  private addInfiniteScrollListeners() {
+    this.bind.on(<HTMLElement>this.options.infiniteScrollContainer, 'scroll', () => {
+      this.successiveScrollCount = 0;
+      this.handleScrollOfResultList();
+    });
+    this.bind.onRootElement(QueryEvents.deferredQuerySuccess, () => {
+      setTimeout(() => {
+        this.scrollToTopIfEnabled();
+      }, 0);
+    });
+  }
 
-    this.setupTemplatesVersusLayouts();
-    $$(this.root).on(ResultLayoutEvents.populateResultLayout, (e, args: IResultLayoutPopulateArgs) =>
-      args.layouts.push(this.options.layout)
-    );
-    this.setupRenderer();
-    this.makeElementFocusable();
-    this.ensureHasId();
+  private addPagerListeners() {
+    this.bind.onQueryState(MODEL_EVENTS.CHANGE_ONE, QUERY_STATE_ATTRIBUTES.FIRST, () => {
+      this.bind.oneRootElement(QueryEvents.deferredQuerySuccess, () => {
```

Comment:
not the `oneRootElement` before it was adding a listener on every page change

Reply:
Could we use the same `oneRootElement` when the result list is in infinite scroll mode? If so, then we can extract the listener into a function and reuse it.

Reply:
No they're explicitly different, this is why the "ONCE" was in caps in the test 😄